langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockChatModelIT.java

```diff
@@ -78,6 +96,445 @@ void testBedrockAnthropicV3SonnetChatModelImageContent() {
+
+        AiMessage aiMessage = response.content();
+        assertThat(aiMessage.text()).isNull();
+        assertThat(aiMessage.toolExecutionRequests()).hasSize(2);
+
+        ToolExecutionRequest firstToolExecutionRequest = aiMessage.toolExecutionRequests().get(0);
+        assertThat(firstToolExecutionRequest.id()).isNotBlank();
+        assertThat(firstToolExecutionRequest.name()).isEqualTo("calculator");
+        assertThat(firstToolExecutionRequest.arguments()).isEqualToIgnoringWhitespace("{\"first\": 2, \"second\": 2}");
+
+        ToolExecutionRequest secondToolExecutionRequest = aiMessage.toolExecutionRequests().get(1);
+        assertThat(secondToolExecutionRequest.id()).isNotBlank();
+        assertThat(secondToolExecutionRequest.name()).isEqualTo("calculator");
+        assertThat(secondToolExecutionRequest.arguments()).isEqualToIgnoringWhitespace("{\"first\": 3, \"second\": 3}");
+
+        TokenUsage tokenUsageCalc = response.tokenUsage();
+        assertThat(tokenUsageCalc.inputTokenCount()).isGreaterThan(0);
+        assertThat(tokenUsageCalc.outputTokenCount()).isGreaterThan(0);
+        assertThat(tokenUsageCalc.totalTokenCount())
+                .isEqualTo(tokenUsageCalc.inputTokenCount() + tokenUsageCalc.outputTokenCount());
+
+        assertThat(response.finishReason()).isEqualTo(TOOL_EXECUTION);
+
+        ToolExecutionResultMessage firstToolExecutionResultMessageCalc = from(firstToolExecutionRequest, "4");
+        ToolExecutionResultMessage secondToolExecutionResultMessageCalc = from(secondToolExecutionRequest, "6");
+        List<ChatMessage> messages = asList(
+                userMessage,
+                aiMessage,
+                firstToolExecutionResultMessageCalc,
+                secondToolExecutionResultMessageCalc
+        );
+
+        Response<AiMessage> secondeResponse = bedrockChatModel.generate(messages, toolSpecifications);
+
+        AiMessage secondAiMessage = secondeResponse.content();
+        assertThat(secondAiMessage.text()).contains("4", "6");
+        assertThat(secondAiMessage.toolExecutionRequests()).isNull();
+
+        TokenUsage secondTokenUsage = secondeResponse.tokenUsage();
+        assertThat(secondTokenUsage.inputTokenCount()).isGreaterThan(0);
+        assertThat(secondTokenUsage.outputTokenCount()).isGreaterThan(0);
+        assertThat(secondTokenUsage.totalTokenCount())
+                .isEqualTo(secondTokenUsage.inputTokenCount() + secondTokenUsage.outputTokenCount());
+
+        assertThat(secondeResponse.finishReason()).isEqualTo(STOP);
+    }
+
+    @Test
+    void testNoParametersFunctionCallingWithBedrockAnthropicV3SonnetChatModel() {
+
+        BedrockAnthropicMessageChatModel bedrockChatModel = BedrockAnthropicMessageChatModel
+                .builder()
+                .temperature(0.50f)
+                .maxTokens(300)
+                .region(Region.US_EAST_1)
+                .model(BedrockAnthropicMessageChatModel.Types.AnthropicClaude3_5SonnetV1.getValue())
```

Comment:
```suggestion
                .model(BedrockAnthropicMessageChatModel.Types.AnthropicClaude3SonnetV1.getValue())
```
I don't have access to 3.5 :(